modules/azurerm_vm/main.tf

```diff
@@ -7,10 +7,11 @@ resource "azurerm_public_ip" "vm_ip" {
   name                = "${var.name}-public_ip"
   location            = var.azure_rg_location
   resource_group_name = var.azure_rg_name
-  allocation_method   = "Static"
+  allocation_method   = var.public_ip_allocation_method
```

Comment:
code would be cleaner like this
```terraform
create_static_ip = var.create_nfs_static_ip
```
Not quite sure why it needs to be spelled out. Very wordy and extra for a simple flag.

Reply:
The terraform module needs either Static or Dynamic. So if we created a boolean flag for the end user it would need to be something more like:

allocation_method   = var.create_nfs_static_ip ? "Static" : "Dynamic"